### DISCUSSION TITLE
Fetch vs. Axios

### DISCUSSION THREAD

**Discussion by devtobi** [it-at-m/refarch-templates#128] [General]
At the moment, we are using standard JS fetch-api for requests. Should we maybe change fetch for axios. What would be the benefits of switching?

**Comment by langehm** [comment]
If apply some ten-stack style functions around the request itself we do not need any more library.

**Comment by devtobi** [comment]
As far as I found axios has the following benefits compared to fetch-API:
- Automatic JSON (de)serialization leads to a more concise usage
- More unified error handling (fetch only handles 404 as error and thus not all problems can be catched via 'catch' mechanism)
- Axios allows simple abortion of long running requests (or combined requests) compared to complicated AbortController in fetch
- Axios supports timeout functionality, whereas fetch needs to manually use AbortController for this)
- Axios allows configuration of a global configuration object. This would allow us to centralize and reuse the current FetchUtils functionality
- Axios has much better support for form data and multipart uploads

Cons are of course:
- Additional dependency (but very frequently updated)
- No more easy full control over the networking code (however this can also be a benefit)

Additional ideas:
- Combined with the axios-Integration from VueUse TanStack query like API can be implemented even easier (we already have VueUse in place so this would be a direct win)
